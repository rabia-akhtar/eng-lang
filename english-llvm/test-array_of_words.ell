
int test2(string a, int words){

  int i = 0;
  int len = strlen(a);
  int start = 0;
  string b;
  string temp = calloc(1,30);
  int j;
  int w = 0;
  string [] word_s = [""];
  int [] freq = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0];
  int indice =0;
  int present = 0;
  int k;
  int h;
  print_string("FIRST PRINT");
  print(freq{|0|});
  print(freq{|100|});

  for(i = 0; i<len; i++){
    b = calloc(1,2);
    b = string_at(a,i,b);
    if(strcmp(b, " ") == 0){
      for(j = start; j <i; j++){
      b = string_at(a,j,b);
        if(strcmp(b,".") != 0){
        strcat(temp, b);
        }
      }

      k = 0;
      while(k<indice && present== 0){
          print_string("HERE WE ARE");
          print(k); 
          print_string(word_s{|k|});
          if(strcmp(word_s{|k|}, temp) == 0){
          print_string("YES");
          
          print_string("THIS IS A NUMBER");
          print(freq{|k|});
          present = 1;
          print_string("FIRST PRINT");
          print(freq{|0|});
          }

          k++;
      }

      print_string(temp);
      if(present == 0){
      word_s[indice] = calloc(1,20);
      freq[indice] = 1;
      strcpy(word_s{|indice|}, temp);
      print_string("PUT IN ARRAY");
      print(indice);
      print_string(word_s{|indice|});
      indice++;
      }
      present = 0;
      free(temp);
      temp = calloc(1,30);
      start = i+1;
    }

    free(b);
    print_string("FIRST PRINT");
    print(freq{|0|});
    }
    print_string("FIRST PRINT");
    print(freq{|0|});


    print_string("FINAL PRINT");
    for(h = 0; h<indice; h++){

    print_string("index");
    print_string(word_s{|h|});
    print(h);
    print(freq{|h|});

  }



  return 0;
}

int main() {
  string [] test;
  int i;
  string g = "hi emily bau emily candace kim emily";
  test = ["hi","hello","there"];
  i = word_count(g);
  print_string("THIS IS NEW");
  test2(g, i);
  return 0;
}
