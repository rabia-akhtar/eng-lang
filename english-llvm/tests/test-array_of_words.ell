/* this demo code reads in a text document and finds the topic words in the file */

/* function builds array of topic words */
int main_topics(string a){

  /* set variables */
  int i;
  int len = strlen(a);
  int start = 0;
  string x;
  string b;
  string temp;
  string [] topics = [""];
  int j;
  int w = 0;
  int indice =0;
  int present = 0;
  int k;  
  int h;
  int l;

/* iterate through string by character */
  for(i = 0; i<len; i++){
    temp = calloc(30,1);
    x = string_at(a,i,30, 1);

    /* if end of word, build previous word */
    if(strcmp(x, " ") == 0){
      j = start;
      for(; j < i; j++){
        b = string_at(a,j, 30, 1);
        if(strcmp(b,".") != 0){
        strcat(temp, b);
        }
        free(b);
      }

      /* put word in array if not a stop word*/
      if(is_stop_word(temp) == 0){
      topics [indice] = calloc(30, 1);
      strcpy(topics{|indice|}, temp); 
      indice++;
      }
      start = i+1;
    }
    free(x); 
  }

    /* print array of topic words */
    for(h = 0; h < indice; h++){
      print_string(topics{|h|});
      free(topics{|h|});
    }

    free(temp);   
    return 0;
}

/* function takes in filepath, reads in text and returns string */
string read_Essay(string file){
   string s1 = calloc(1, 2000);
   file_ptr fp = open(file, "rb");
   int size = read(s1, 1, 2000, fp);
   close(fp);
   return s1;
}

int main() {
  int i;
  string essay;
  essay = read_Essay("tests/essay1.txt");
  main_topics(essay);
  free(essay);
  return 0;
}