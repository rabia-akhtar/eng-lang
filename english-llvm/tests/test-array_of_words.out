Oceans
lakes
much
common
quite
different
well
Both
bodies
water
oceans
large
bodies
salt
water
lakes
much
smaller
bodies
fresh
water
Lakes
usually
surrounded
land
oceans
surround
